ZFSin/spl/include/sys/systm.h

```diff
@@ -39,7 +39,9 @@ static inline void bsd_untimeout(void(*func)(void *), void *ID)
 	struct bsd_timeout_wrapper *btw = (struct bsd_timeout_wrapper *)ID;
 	LARGE_INTEGER p = { -1 };
 	btw->init = 0;
-	KeSetTimer(&btw->timer, p, NULL);
+	ASSERT(btw->init == BSD_TIMEOUT_MAGIC); // timer was not initialized 
+	if(btw->init == BSD_TIMEOUT_MAGIC)
+		KeSetTimer(&btw->timer, p, NULL);
```

Comment:
I wrote the bsdtimeout code and then found I couldn't unload the driver to test unloading it, so I'm not surprised the unload needs fixes. But I think I'll try to find a way that doesn't use the "debug build" init magic value. I'll put it on my todo for a quiet day :)   (Especially since we set `init` to 0 just before the ASSERT

Reply:
oh, just had one of that "what the heck" moments 😂 Guess I just overlooked that `btw->init = 0;` the first time, but I would say leaving an assertion as a reminder here doesn't hurt though ...